wdl-engine/src/eval.rs

```diff
@@ -149,3 +201,155 @@ impl<'a> ScopeRef<'a> {
+    }
+
+    /// Gets the working directory of the evaluated task.
+    pub fn work_dir(&self) -> &Path {
+        &self.work_dir
+    }
+
+    /// Gets the temp directory of the evaluated task.
+    pub fn temp_dir(&self) -> &Path {
+        &self.temp_dir
+    }
+
+    /// Gets the command file of the evaluated task.
+    pub fn command(&self) -> &Path {
+        &self.command
+    }
+
+    /// Gets the stdout value of the evaluated task.
+    pub fn stdout(&self) -> &Value {
+        &self.stdout
+    }
+
+    /// Gets the stderr value of the evaluated task.
+    pub fn stderr(&self) -> &Value {
+        &self.stderr
+    }
+
+    /// Gets the outputs of the evaluated task.
+    ///
+    /// This is `Ok` when the task executes successfully and all of the task's
+    /// outputs evaluated without error.
+    ///
+    /// Otherwise, this contains the error that occurred while attempting to
+    /// evaluate the task's outputs.
+    pub fn outputs(&self) -> &EvaluationResult<Outputs> {
+        &self.outputs
+    }
+
+    /// Converts the evaluated task into an evaluation result.
+    ///
+    /// Returns `Ok(_)` if the task outputs were evaluated.
+    ///
+    /// Returns `Err(_)` if the task outputs could not be evaluated.
+    pub fn into_result(self) -> EvaluationResult<Outputs> {
+        self.outputs
+    }
+
+    /// Handles the exit of a task execution.
+    ///
+    /// Returns an error if the task failed.
+    fn handle_exit(&self, requirements: &HashMap<String, Value>) -> anyhow::Result<()> {
+        let mut error = true;
+        if let Some(return_codes) = requirements
+            .get("return_codes")
+            .or_else(|| requirements.get("returnCodes"))
+        {
+            match return_codes {
+                Value::Primitive(PrimitiveValue::String(s)) if s.as_ref() == "*" => {
+                    error = false;
+                }
+                Value::Primitive(PrimitiveValue::String(s)) => {
+                    bail!("invalid return code value `{s}`; only `*` is accepted");
+                }
+                Value::Primitive(PrimitiveValue::Integer(ok)) => {
+                    if self.status_code == i32::try_from(*ok).unwrap_or_default() {
+                        error = false;
+                    }
+                }
+                Value::Compound(CompoundValue::Array(codes)) => {
+                    error = !codes.as_slice().iter().any(|v| {
+                        v.as_integer()
+                            .map(|i| i32::try_from(i).unwrap_or_default() == self.status_code)
+                            .unwrap_or(false)
+                    });
+                }
+                _ => unreachable!("unexpected return codes value"),
+            }
+        } else {
+            error = self.status_code != 0;
+        }
+
+        if error {
+            bail!(
+                "task process has terminated with status code {code}; see standard error output \
+                 `{stderr}`",
```

Comment:
Perhaps just include the path to the test's execution directory and mention that it contains the stdout, stderr, and working directory for the task? Otherwise it might be path overload.

Reply:
Path overload is a valid concern, but on the other hand, I like being able to quickly doubleclick to select each path and paste it into my terminal. Maybe just the execution dir is enough though... I'm not sure.

Is stderr, stdout, work dir, and temp dir, all at the root of the execution dir? If so, I think probably just the exec dir is enough. If `ls <exec dir>` immediately gives me everything I'm looking for, it would be ok.

Reply:
They are all at the root of the execution dir (`stdout`, `stderr`, `command`, `work_dir`, and `temp_dir`).

Reply:
Very nice. That's as it should be IMO, but there are unspecified runners out there that make tracking those down difficult by being nested/in weird directories, etc.

In this case, I would change the message to just print the execution dir path 🚀